### ISSUE TITLE
Redundant expression false positive

### ISSUE THREAD

**Issue by rjerue** [clj-kondo/clj-kondo#1185]
**version**

v2021.02.28 / RELEASE

**platform**

JVM

**editor**

VS Code, though it's not coming from the editor.

**problem**

Redundant keys warning triggers when deconstruction is done across two lines.

**repro**

```clojure
(defn fun
  [{:keys [a-key]}]
  (when-let [{:keys [a-deep]} a-key])
  (println a-key))
```

```bash
clojure -A:kondo --lint src/example.cljs
WARNING: When invoking clojure.main, use -M
src/example.clj:5:15: warning: Redundant expression: :keys
linting took 127ms, errors: 0, warnings: 1
```
**config**

I don't think this is related to the config.

**expected behavior**

No warning

**Comment by borkdude** [comment]
Thanks. You can disable the `:redundant-expression` or use a `#_:clj-kondo/ignore` annotation temporarily until this is fixed.

**Comment by reify-rjerue** [comment]
Just a note that I did mess up the above impl, I _meant_ to say

```
(defn fun
  [{:keys [a-key]}]
  (when-let [{:keys [a-deep]} a-key])
  (println a-deep))
```

Still has the same error though

**Comment by rjerue** [comment]
Just a note that I did mess up the above impl, I _meant_ to say

```
(defn fun
  [{:keys [a-key]}]
  (when-let [{:keys [a-deep]} a-key])
  (println a-deep))
```

Still has the same error though

**Comment by cjsauer** [comment]
As noted in #298, I'm also seeing false positives regarding `(js/...)` interop expressions that take constant `"string"` args.

**Comment by borkdude** [comment]
@cjsauer Your issue is solved in #1183
